app/views/mainpages/_content.html.erb

```diff
@@ -21,14 +21,10 @@
               <p><%= result.description %></p>
             </div>
             <div class="coupon_opc">
-              <%= button_to "delete", {:controller => :cupons, :action => 'destroy', :id => result.id }, :method => :delete %>
-              <%= link_to "Edit", edit_cupon_path(result)%>
+              <%= link_to "Ver", cupon_path(result.id) %>
+              <%= ((button_to "delete", {:controller => :cupons, :action => 'destroy', :id => result.id }, :method => :delete) if current_user.id == result.user_id) if current_user %>
+              <%= ((link_to "Edit", edit_cupon_path(result)) if current_user.id == result.user_id) if current_user %>
```

Comment:
Also I think that's a lot of nested ifs in the view I recommend trying to charge the controller with the logic

Reply:
I reduced it as much as possible but if I modify it more for some reason the delete button disappears, that's the only way it renders and works correctly. :(